slack-lint-checks/src/main/java/slack/lint/rx/RxObservableEmitDetector.kt

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2025 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package slack.lint.rx
+
+import com.android.tools.lint.checks.DataFlowAnalyzer
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.ULambdaExpression
+import slack.lint.util.sourceImplementation
+
+class RxObservableEmitDetector : Detector(), SourceCodeScanner {
+  override fun getApplicableUastTypes() = listOf(UCallExpression::class.java)
+
+  override fun createUastHandler(context: JavaContext): UElementHandler {
+    return object : UElementHandler() {
+      override fun visitCallExpression(node: UCallExpression) {
+        val issue = functionToIssue[node.methodName] ?: return
+        val lambdaExpression = node.valueArguments.lastOrNull() as? ULambdaExpression ?: return
+        val producerScopeParam = lambdaExpression.parameters.firstOrNull() ?: return
+
+        // Verify the parameter is a ProducerScope
+        if (!producerScopeParam.type.canonicalText.startsWith(PROVIDER_SCOPE_FQN)) return
+
+        var sendCalled = false
+
+        val visitor =
+          object : DataFlowAnalyzer(emptySet()) {
+            override fun visitCallExpression(node: UCallExpression): Boolean {
+              // If we find a nested factory method, return immediately and stop traversing this
+              // code path.
+              // Note: this factory will be validated by the UElementHandler above
+              if (node.methodName in functionToIssue) return true
```

Comment:
Would be good to add a test with an import alias just to ensure this isn't defeated by that

Reply:
Good idea

Reply:
https://github.com/slackhq/slack-lints/pull/375/commits/10c9549e02e36c5c8b7917f576aa88e9961ef3c3